Core/Meta/Type.cs

```diff
@@ -70,4 +71,77 @@ public DefinedType(string name, Span span, string asString) : base(span, asStrin
             Name = name;
         }
     }
+
+    public static class TypeExtensions
+    {
+        public static bool IsFixedScalar(this TypeBase type)
+        {
+            return type is ScalarType and { BaseType: not BaseType.String };
+        }
+        public static bool IsStruct(this TypeBase type, ISchema schema)
+        {
+            return type switch
+            {
+                DefinedType dt when schema.Definitions[dt.Name] is StructDefinition => true,
+                _ => false
+            };
```

Comment:
This can be `return type is DefinedType dt && schema.Definitions[dt.Name] is StructDefinition`, I think. Same for the similar methods below.